src/lib/comms/sol-lwm2m.c

```diff
@@ -3237,32 +3562,33 @@ obj_ctx_clear(struct sol_lwm2m_client *client, struct obj_ctx *ctx)
 }
 
 static void
-server_connection_ctx_clear(struct server_conn_ctx *conn_ctx)
+server_connection_ctx_clear(struct server_conn_ctx *conn_ctx, bool is_bootstrap)
 {
     if (conn_ctx->pending_pkt)
         sol_coap_packet_unref(conn_ctx->pending_pkt);
     if (conn_ctx->hostname_handle)
         sol_network_hostname_pending_cancel(conn_ctx->hostname_handle);
     sol_vector_clear(&conn_ctx->server_addr_list);
-    free(conn_ctx->location);
+    if (!is_bootstrap)
+        free(conn_ctx->location);
```

Comment:
Wow, I really believe I was getting some compiler warning/error... But I just tried removing it and all is well. oO

Anyway, fixed. Thanks!